src/server/service/batch.rs

```diff
@@ -130,9 +132,8 @@ fn future_batch_get_command<E: Engine, L: LockManager>(
         GRPC_MSG_HISTOGRAM_STATIC
             .kv_batch_get_command
             .observe(begin_instant.elapsed_secs());
-        Ok(())
-    });
-    poll_future_notify(f);
+    };
+    poll_future_notify(Compat::new(f.unit_error().boxed()));
```

Comment:
Could it be implement with `futures 0.3`?

Reply:
I think so. But the batch mechanism is quite a lot of code. It definitely deserves a separate PR.